clients/src/main/java/org/apache/kafka/clients/consumer/internals/DefaultFetcher.java

```diff
@@ -130,7 +130,7 @@
  *     on a different thread.</li>
  * </ul>
  */
-public class Fetcher<K, V> implements Closeable {
+public class DefaultFetcher<K, V> implements Closeable {
```

Comment:
I'm thinking if we aren't implementing an interface, do we actually need to rename the class? Also because the implementation would be significantly different from the current fetcher, I wonder if there will be a common interface to implement.

Reply:
There is a common set of methods to implement.

The reason this is being done this way is to avoid a large set of diffs. 

Steps:

1. Rename `Fetcher` to `DefaultFetcher`
2. Extract the common methods into an interface named `Fetcher`
3. Retcon `DefaultFetcher` (the original `Fetcher`) to implement the `Fetcher` interface
4. Add the `ProtoFetcher` (for use by the new, refactored consumer`) to implement the `Fetcher` interface
5. Generalize the fetcher tests to use the interface and add subclasses that test the specific implementations